specs/spec_reader.go

```diff
@@ -8,78 +8,93 @@ import (
 )
 
 type SpecReader struct {
-	sources      map[string]Source
-	destinations map[string]Destination
+	Sources      map[string]*Source
+	Destinations map[string]*Destination
 }
 
-func NewSpecReader(directory string) (*SpecReader, error) {
-	reader := SpecReader{
-		sources:      make(map[string]Source),
-		destinations: make(map[string]Destination),
-	}
-	files, err := os.ReadDir(directory)
+func (r *SpecReader) loadSpecsFromFile(path string) error {
+	data, err := os.ReadFile(path)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read directory %s: %w", directory, err)
+		return fmt.Errorf("failed to read file %s: %w", path, err)
 	}
+	data = []byte(os.ExpandEnv(string(data)))
 
-	configFilesCount := 0
-	for _, file := range files {
-		if !file.IsDir() && !strings.HasPrefix(file.Name(), ".") && strings.HasSuffix(file.Name(), ".yml") {
-			data, err := os.ReadFile(filepath.Join(directory, file.Name()))
-			if err != nil {
-				return nil, fmt.Errorf("failed to read file %s: %w", file.Name(), err)
-			}
-			var s Spec
-			if err := SpecUnmarshalYamlStrict(data, &s); err != nil {
-				return nil, fmt.Errorf("failed to unmarshal file %s: %w", file.Name(), err)
+	// support multiple yamls in one file
+	for i, doc := range strings.Split(string(data), "---") {
```

Comment:
Bit of an edge case to be sure, but if some string in the config happened to contain `---`, that would cause the file to be broken up incorrectly. Might be better to split by lines first, then look for a line that only contains `---`

Reply:
fixed